### ISSUE TITLE
Invoke-IcingaCheckUsedPartitionSpace - problem with mirrors disks

### ISSUE THREAD

**Issue by puseidr** [Icinga/icinga-powershell-plugins#187]
Hi, 
the cmdlet Invoke-IcingaCheckUsedPartitionSpace throws exceptions when you have mirrored disks.

(even with version 1.5.0)

```
PS C:\Windows\system32> Invoke-IcingaCheckUsedPartitionSpace
Ausnahme beim Aufrufen von "Add" mit 2 Argument(en):  "Das Element wurde bereits hinzugefügt. Schlüssel im Wörterbuch:
"3". Hinzuzufügender Schlüssel: "3"."
In C:\Program
Files\WindowsPowerShell\Modules\icinga-powershell-plugins\provider\disks\Get-IcingaDiskPartitionAssignment.psm1:53
Zeichen:9
+         $PartitionInformation[$DiskId].Partitions.Add(
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ArgumentException

Ausnahme beim Aufrufen von "Add" mit 2 Argument(en):  "Das Element wurde bereits hinzugefügt. Schlüssel im Wörterbuch:
"3". Hinzuzufügender Schlüssel: "3"."
In C:\Program
Files\WindowsPowerShell\Modules\icinga-powershell-plugins\provider\disks\Get-IcingaDiskPartitionAssignment.psm1:53
Zeichen:9
+         $PartitionInformation[$DiskId].Partitions.Add(
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ArgumentException

[OK] Used Partition Space: 4 Ok
| 'partition_d'=656880300000B;;;0;852476900000 'partition_c'=62066870000B;;;0;107375200000
```

![image](https://user-images.githubusercontent.com/2630551/120603279-227b5a00-c44c-11eb-98b5-ea7b3b509b60.png)

I think it's the same problem like issue #130

**Comment by LordHepipud** [comment]
Thank you for the issue. Can you please test the linked PR and check if this resolves the issue and all disks/partitions you expect to be there, are present?

**Comment by puseidr** [comment]
the PR removed the exceptions and it looks like its working but not all partitions are listed:

```
PS C:\Windows\system32> Invoke-IcingaCheckUsedPartitionSpace -Verbosity 3
[OK] Used Partition Space: 4 Ok (All must be [OK])
\_ [OK] Partition C: 57.81GiB
\_ [OK] Partition C: 57.81GiB
\_ [OK] Partition D: 611.83GiB
\_ [OK] Partition D: 611.83GiB
| 'partition_d'=656948300000B;;;0;852476900000 'partition_c'=62075750000B;;;0;107375200000
```

Partition E is missing! 

on another server the same, it only shows C and D and all other partition (E,F and G) are not shown by the cmdlet output/performacedata

```
PS C:\Windows\system32> Get-Volume

DriveLetter FileSystemLabel FileSystem DriveType HealthStatus OperationalStatus SizeRemaining      Size
----------- --------------- ---------- --------- ------------ ----------------- -------------      ----
D           Volume          NTFS       Fixed     Healthy      OK                     182.1 GB    200 GB
E           VM              NTFS       Fixed     Healthy      OK                    358.04 GB 593.93 GB
C           System          NTFS       Fixed     Healthy      OK                     42.19 GB    100 GB
```

and BTW: i'm missing the percentage usage for the partitions in the performancedata (in 1.4.1 it showed the percentage and byte values), is there a reason why you removed this or is it a bug?

**Comment by LordHepipud** [comment]
Thanks for the input and the test. Can you please run the following command

```powershell
Get-IcingaWindowsInformation Win32_LogicalDiskToPartition;
```

and post the result? I need to setup a test environment with mirrored disks in the meantime, because I have no system available for this yet.

Did you use windows internal tools to mirror the disks and create the raid?

For the percentage values: They were removed while we redesigned the plugin. Icinga for Windows 1.5.0 allows for most plugins to add plain value thresholds, like '10GB' or % values like '10%'.

The performance data is always set properly, allowing to do calculations directly within the graphing solution like Grafana.

**Comment by puseidr** [comment]
here is the output:

```
PS C:\Windows\system32> Get-IcingaWindowsInformation Win32_LogicalDiskToPartition;


Antecedent      : Win32_DiskPartition (DeviceID = "Disk #0, Partition #2")
Dependent       : Win32_LogicalDisk (DeviceID = "C:")
EndingAddress   : 107720212479
StartingAddress : 344981504
PSComputerName  :

Antecedent      : Win32_DiskPartition (DeviceID = "Disk #1, Partition #2")
Dependent       : Win32_LogicalDisk (DeviceID = "C:")
EndingAddress   : 107720212479
StartingAddress : 344981504
PSComputerName  :

Antecedent      : Win32_DiskPartition (DeviceID = "Disk #0, Partition #3")
Dependent       : Win32_LogicalDisk (DeviceID = "D:")
EndingAddress   : 322468577279
StartingAddress : 107720212480
PSComputerName  :

Antecedent      : Win32_DiskPartition (DeviceID = "Disk #1, Partition #3")
Dependent       : Win32_LogicalDisk (DeviceID = "D:")
EndingAddress   : 322468577279
StartingAddress : 107720212480
PSComputerName  :

Antecedent      : Win32_DiskPartition (DeviceID = "Disk #0, Partition #3")
Dependent       : Win32_LogicalDisk (DeviceID = "E:")
EndingAddress   : 960195723263
StartingAddress : 322468577280
PSComputerName  :

Antecedent      : Win32_DiskPartition (DeviceID = "Disk #1, Partition #3")
Dependent       : Win32_LogicalDisk (DeviceID = "E:")
EndingAddress   : 960195723263
StartingAddress : 322468577280
PSComputerName  :
```

the mirroring was done with **diskmgmt.msc** - no fancy raid1 mirroring.. the server doesn't have a raid controller :-/

**Comment by wrkhunte** [comment]
I have the same problem with a dynamic disk on which multiple volumes are created.

My dynamic disk looks something like this:
![image](https://user-images.githubusercontent.com/84439716/124595959-f8112800-de61-11eb-837b-9c05e935c6e7.png)

This is my exception: 
```
Exception calling "Add" with "2" argument(s): "Item has already been added. Key in dictionary: '1'  Key being added: '1'"
At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-plugins\provider\disks\Get-IcingaDiskPartitionAssignment.psm1:53 char:13
+             $PartitionInformation[$DiskId].Partitions.Add(
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ArgumentException

```

I use the Powershell plugins v. 1.5.0

**Comment by LordHepipud** [comment]
This is fixed with v1.7.0
